objects/src/assets/mod.rs

```diff
@@ -8,10 +8,12 @@ use core::{fmt, ops::Deref};
 ///
 /// All assets are encoded using a single word (4 elements) such that it is easy to determine the
 /// type of an asset both inside and outside Miden VM. Specifically:
-/// - The first bit of a fungible asset is always ONE, and the last 32-bits of the 3rd element
-///   are always set to ZEROs.
-/// - The first bit of a non-fungible asset is always ZERO, and the last 32-bits of the 3rd element
-///   are always set to 2^31 (i.e., ONE followed by 31 ZEROs).
+///   Element 1 will be:
+///    - ZERO for a fungible asset
+///    - non-ZERO for a non-fungible asset
+///   The most significant bit will be:
+///    - ONE for a fungible asset
+///    - ZERO for a non-fungible asset
 ///
 /// The above properties guarantee that there can never be a collision between a fungible and a
 /// non-fungible asset. Collision resistance of both fungible and non-fungible assets is ~110 bits.
```

Comment:
I think we should re-write this paragraph.

Reply:
done